x/protorev/types/genesis.go

```diff
@@ -0,0 +1,74 @@
+package types
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// InputAmountList contains a list of input amounts to test when creating the optimal amount to swap in the
+// binary search method
+var InputAmountList []sdk.Int
+
+// AtomDenomination stores the native denom name for Atom on chain used for route building
+var AtomDenomination string = "ibc/27394FB092D2ECCD56123C74F36E4C1F926001CEADA9CA97EA622B25F41E5EB2"
```

Comment:
Should this be hard-coded as such? What about an on-chain parameter?

Reply:
These two denoms must be known by module at run-time since the trading executed by protorev will always start and end with either atom or osmo. We can configure this so that we initialize this in the genesis state instead of having it be a hardcoded value.

Reply:
IMO, probably better in this case for it to be hardcoded, as we don't want it to use anything other than osmo or atom, for the foreseeable future